lib/internal/vm/noderealm.js

```diff
@@ -0,0 +1,115 @@
+'use strict';
+
+// NodeRealm was originally a separate module developed by
+// Anna Henningsen and published separately on npm as the
+// synchronous-worker module under the MIT license. It has been
+// incorporated into Node.js with Anna's permission.
+// See the LICENSE file for LICENSE and copyright attribution.
+
+const {
+  Promise,
+} = primordials;
+
+const {
+  NodeRealm: NodeRealmImpl,
+} = internalBinding('contextify');
+
+const EventEmitter = require('events');
+const { setTimeout } = require('timers');
+const { pathToFileURL } = require('url');
+
+let debug = require('internal/util/debuglog').debuglog('noderealm', (fn) => {
+  debug = fn;
+});
+
+class NodeRealm extends EventEmitter {
+  #handle = undefined;
+  #process = undefined;
+  #global = undefined;
+  #stoppedPromise = undefined;
+  #loader = undefined;
+
+  /**
+   */
+  constructor() {
+    super();
```

Comment:
Can we please add an `emitExperimentalWarning`? I can imagine that there will be a point when we want to drop the flag but keep the warning for a while.